### DISCUSSION TITLE
AttachmentEntry list

### DISCUSSION THREAD

**Discussion by vlsitor** [nextgenhealthcare/connect#5686] [Q&A] [answered]
Hello,

I am using a SMTP Sender connector and I try to send attachments via e-mail.
In the transformer of my source connector I gather a variable amount of base64 encoded jpg images per message.

For that I create a list that contains AttachmentEntry values.

`var attachmentList = Lists.list();`
`attachmentList.add(new AttachmentEntry('name.jpg', base64img, 'image/jpg'));`
`$c('attachmentList',attachmentList);`

On the destination connecter I put this list to the Use List parameter.

![image](https://user-images.githubusercontent.com/125638890/219574405-e47ce9cd-74e7-4f0c-95e5-813416cb670e.png)

Now my problem is, that I receive the mails, but without an attachment.
I already put the attachment list to the logger to see if everything is filled correctly. And for me it seems correct.

**Comment by mmrqa** [comment, marked as answer]
Hi @vlsitor,

I had the same issue, but I found the following post: 

https://forums.mirthproject.io/forum/mirth-connect/support/183146-how-to-handle-file-reader-sftp-source-to-smtp-destination-for-multiple-files

Seems like the variable should **not be written as velocity syntax, so just "attachmentList" without "${}".** 
I tried it and it worked out for me. 

Maybe it helps for your next project :)
